server/src/main/java/org/opensearch/index/store/remote/filecache/FileCache.java

```diff
@@ -123,4 +133,35 @@ public CacheUsage usage() {
     public CacheStats stats() {
         return theCache.stats();
     }
+
+    /**
+     * Restores the file cache instance performing a folder scan of the
+     * {@link org.opensearch.index.store.remote.directory.RemoteSnapshotDirectoryFactory#LOCAL_STORE_LOCATION}
+     * directory within the provided file cache path.
+     * @param fileCacheNodePath path pointing to the file cache directory
+     * @param fileCache the file cache instance to be restored
+     */
+    /* TODO: This process is best served with a ForkJoin mechanism. Add support for a
+        {@link java.util.concurrent.RecursiveTask} once support for {@link java.util.concurrent.ForkJoinPool}
+        has been added within OpenSearch */
+    public static void restoreFileCache(final NodeEnvironment.NodePath fileCacheNodePath, final FileCache fileCache) {
+        try {
+            List<Path> cacheDataPaths = collectFileCacheDataPath(fileCacheNodePath);
+            for (Path cacheDataPath : cacheDataPaths) {
+                if (Files.isDirectory(cacheDataPath) && Files.isDirectory(cacheDataPath.resolve(LOCAL_STORE_LOCATION))) {
+                    File subPathFolder = cacheDataPath.resolve(LOCAL_STORE_LOCATION).toFile();
+                    for (File subPathFile : subPathFolder.listFiles()) {
+                        if (subPathFile.isFile()) {
+                            fileCache.put(
+                                subPathFile.toPath().toRealPath(),
+                                new FileCachedIndexInput.ClosedIndexInput(subPathFile.length())
```

Comment:
This is so gross, but I understand why you're doing it. I think we should be caching a pointer to a chunk on disk instead of an InputIndex (we can create an IndexInput for the file chunk on demand when it is needed) but that is a bigger cache refactoring discussion.

Reply:
Should we create an issue to track this?